File: 07_JavaScript/Exercices/exo35/classes/Guerrier.js
import Personnage from "./Personnage.js";

export default class Guerrier extends Personnage {
    armure;

    constructor (nom, vie, force, armure) {
        super (nom, vie, force);
        this.armure = armure;
        this.vie += this.armure;
    }

    attaquer(personnage) {
        personnage.vie -= (this.force + this.armure);
        console.log(`${this.nom} inflige ${this.force + this.armure} à ${personnage.nom}!`);
    }
    
}

File: 19_Java/03_DESIGN_PATTERN/exercices/src/main/java/org/example/exercice03TpBuilderEtAbstractFactory/entity/personnage/Guerrier.java
package org.example.exercice03TpBuilderEtAbstractFactory.entity.personnage;

public class Guerrier extends Personnage {

    private int armure;



    private Guerrier (GuerrierBuilder guerrierBuilder) {
        super(guerrierBuilder);
        this.armure = guerrierBuilder.armure;
    }

    public static class GuerrierBuilder extends AbstractPersonnageBuilder {

        private int armure;

        public GuerrierBuilder armure (int armure){
            this.armure = armure;
            return this;
        }

        @Override
        public Personnage build() {
            return new Guerrier(this);
        }
    }
}
